### ISSUE TITLE
Can't create a record with nested attribute

### ISSUE THREAD

**Issue by lionelrudaz** [JSONAPI-Resources/jsonapi-resources#354]
Hi guys,

This is probably a newbie question, but I haven't found anything similar in previous issues, so I don't know what I do wrong.

I'm trying to create a Search that has a Category in my API. I'm creating a custom search_controller because I need some specific features on my search API.

Here's the resource:

```
module Api
  module V1
    class SearchResource < JSONAPI::Resource
      attributes :lat, :lon, :radius, :first_doctor, :first_doctor_first_availability, :first_doctor_first_slot
      has_one :category
    end
  end
end
```

And the payload I'm sending:

```
{"data":{"attributes":{"lat":46.6143844,"lon":6.508884500000022,"radius":null},"category":{"data":{"id":"1","attributes":{"title":"General Medicine","description":null,"rank":0,"created-at":null,"updated-at":null},"type":"categories"}},"type":"searches"}}
```

I get the following error:

```
{"errors":[{"title":"Param not allowed","detail":"category is not allowed.","id":null,"href":null,"code":105,"source":null,"links":null,"status":"bad_request"}]}
```

There's probably something really silly I'm doing wrong. Any ideas?

Let me know if you need more information.

Thanks in advance

**Comment by lgebhardt** [comment]
In the payload you should have a `relationships` key that's a sibling of `attributes`. Then the second issue is there category should only be provided as an `id` and `type` without attributes. In other words you are creating a relationship to an existing category, not a category and a search in one request.

See http://jsonapi.org/format/#crud-creating

``` json
{"data": {
  "attributes":{
    "lat":46.6143844,
    "lon":6.508884500000022,
    "radius":null
  },
  "realtionships": {
    "category":{
      "data":{
         "id":"1",
         "type":"categories"
      }
    },
  "type":"searches"
  }
}
```

Of course you will need a CategoryResource defined in the system as well.

**Comment by lionelrudaz** [comment]
Hi Larry,

Thanks for your quick answer. Got you.

I have to check how I can change the call that's done on my EmberJS application so that it can change the payload. 

I'll post the solution when I get it work.

Thanks,

Lionel

**Comment by xdmx** [comment]
I'm having a similar problem, I have the following resources:

```
class ActivityResource < BaseResource
      has_many :comments
end

class CommentResource < BaseResource
      has_one :activity
end
```

And I'm sending the following request:

```
{"data"=>{"type"=>"comments",
                "attributes"=>{"body"=>"Good luck!"},
                "relationships"=>{"activity"=>{"data"=>{"id"=>631898775, "type"=>"activities"}}}}
}
```

But I get as response:

```
{"errors":[{"title":"Param not allowed","detail":"activity is not allowed.","id":null,"href":null,"code":105,"source":null,"links":null,"status":"bad_request"}]}
```

The Activity # 631898775 exists

**Comment by barelyknown** [comment]
@xdmx Is activity in the `creatable_fields` or `updatable_fields` (whichever you're doing) for your resource?

**Comment by xdmx** [comment]
I forgot to add it :smile:

**Comment by barelyknown** [comment]
Great. Glad that's all it was.

**Comment by lionelrudaz** [comment]
Hi Larry, 

Thanks a lot for your help.

I resolved the issue about sending the right data to the API. I had to do two changes:

1) Remove category: {embedded: 'always'} in my search serializer
2) Load the selected category from the store before assigning it to my search (found the solution in the guide [here](http://guides.emberjs.com/v1.13.0/models/creating-and-deleting-records/))

I hope this helps if you face the same issue.

Cheers,

Lionel
